taskchampion/xtask/src/main.rs

```diff
@@ -29,3 +45,99 @@ fn codegen() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+/// `cargo xtask msrv (X.Y)`
+///
+/// This updates all of the places in the repo where the MSRV occurs to <arg2> for the list of file paths defined as const in Xtask main.rs.
+/// If no pre-existing Xtask-formatted MSRV is found, inserts the MSRV as first comment block after the last `//!` library-doc comments.
+fn msrv(args: Vec<String>) -> anyhow::Result<()> {
+    match args.len(){
+        0 ..= 2 => println!("Error: xtask: Command `caro run xtask msrv` expects at least one argument, none detected."),
+        3 => {
+
+                // check that (X.Y) argument is (mostly) valid:
+                for xy in & mut args[2].chars() {
+                    if !xy.is_numeric() && xy != '.'{
+                        anyhow::bail!("xtask: Invalid argument format. Xtask MSRV argument takes the form \"X.Y(y)\", where XYy are numbers. eg: `cargo run xtask MSRV 1.68`");
+                    }
+                }
+
+                // for each file in const paths vector
+                for path in MSRV_FILE_PATHS {
+                    let path : & Path = Path::new(& path);
+
+                    // check if path exists, if not, skip then continue
+                    if ! path.exists() {
+                        println!("xtask: Warning: xtask: Path `{}` not found. Skipping.", path.display());
+                        continue;
+                    }
+
+                    let file : File = File::options().read(true).write(true).open(path).unwrap();
+                    let reader = BufReader::new(&file);
+
+                    // set pattern to search for and the replacement string
+                    let msrv_pattern = "///  MSRV = \"";
+                    let replacement_string = format!("///  MSRV = \"{}\"\n", args[2]);
+
+                    // for each line in file
+                    let mut file_string = String::new();
+                    let mut msrv_pattern_found = false; //should change this to an int and update with the line# when found
+                    let mut comment_pattern_last_offset : usize = 0;
+                    for line in reader.lines() {
+                        let pattern_offset = & line.as_ref().unwrap().find(msrv_pattern);
```

Comment:
A more common idiom here, with line of type `Result<String, Error>`, is to use `?`.  If `line` was just used once, that might be:
```rust
                        let pattern_offset = line?.find(msrv_pattern);
```
but in this case `line` is used a few more times, so just check for errors once:
```rust
                        let line = line?;
                        let pattern_offset = line.find(msrv_pattern);
```

Reply:
Done.